src/openms/source/FORMAT/FASTAFile.cpp

```diff
@@ -39,179 +39,240 @@
 
 #include <OpenMS/CONCEPT/LogStream.h>
 
-#include <seqan/basic.h>
-#include <seqan/stream.h>
-#include <seqan/seq_io/guess_stream_format.h>
-#include <seqan/seq_io/read_fasta_fastq.h>
 
 namespace OpenMS
 {
-  using namespace std;
-  typedef seqan::RecordReader<std::fstream, seqan::SinglePass<> > FASTARecordReader;
-
-  FASTAFile::FASTAFile()
-  : reader_(std::nullptr_t()), // point to nothing
-    entries_read_(0)
-  {
-  }
-  
-  FASTAFile::~FASTAFile()
-  {
-    // infile_ and outfile_ will close automatically when going out of scope. No need to do it explicitly here.
-  }
-
-  void FASTAFile::readStart(const String& filename)
-  {
-    if (!File::exists(filename))
+    using namespace std;
+
+    FASTAFile::FASTAFile()
+            : entries_read_(0)
     {
-      throw Exception::FileNotFound(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, filename);
     }
 
-    if (!File::readable(filename))
+    FASTAFile::~FASTAFile()
     {
-      throw Exception::FileNotReadable(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, filename);
+        // infile_ and outfile_ will close automatically when going out of scope. No need to do it explicitly here.
     }
 
-    if (infile_.is_open()) infile_.close(); // precaution
 
-    infile_.open(filename.c_str(), std::ios::binary | std::ios::in);
+    bool FASTAFile::getLine_(std::istream& is, std::string& t){
+        t.clear();
+        std::istream::sentry se(is, true);
+        if (!se)
+        { // the stream has an error
+            return false;
+        }
+        std::streambuf* sb = is.rdbuf();
+        if(entries_read_== 0)//
+        {
+            if(sb->sgetc() == '>') return false;
+            if(sb->sgetc() == '#')
```

Comment:
thinking about this a bit more:
it's probably faster to skip the PEFF header like it was originally done, because here, you "pay extra" for checking for this header when reading ANY subsequent line. However, at this point, the header is long gone (skipped over).